shell/components/form/GithubPicker.vue

```diff
@@ -1,343 +1,355 @@
+      if (res.message) {
+        this.hasError.branch = true;
+      } else {
+        this.branches = res;
+        this.hasError.branch = false;
       }
     },
-  },
+    onSearch: debounce(function(q) {
+      this.searchForResult(q);
+    }, 1000),
+    status(value) {
+      return !value ? null : 'error';
+    },
+  }
 };
-
 </script>
 
 <template>
   <div class="picker">
-    <div v-if="!hasPrivate && !loadingRecentRepos" class="expand-scope pb-20">
-      <i class="icon icon-info" /> Showing only public repos.  <a
-        href="#"
-        class="text-primary bg-transparent"
-        @click="expandScope"
-      >
-        Click here
-      </a>
-      to grant Rio access to read private repos.
-    </div>
     <div class="row">
-      <div class="col span-4">
-        <LabeledSelect
-          :placeholder="repoPlaceholder"
-          :disabled="loadingRecentRepos"
-          :options="repos"
-          label="full_name"
-          :value="selectedRepo"
-          :clearable="false"
-          @search="onSearchRepos"
-          @input="selectRepo"
-        >
-          <template #spinner="{loading}">
-            <span v-show="loading">
-              <i class="icon icon-spinner icon-spin" /> Loading&hellip;
-            </span>
-          </template>
-          <template #no-options>
-            Type to search GitHub repositories
-          </template>
-          <template slot="option" slot-scope="option">
-            <div class="d-center">
-              <img :src="option.owner.avatar_url" />
-              {{ option.full_name }}
-            </div>
-          </template>
-          <template slot="selected-option" slot-scope="option">
-            <div class="selected d-center">
-              <img :src="option.owner.avatar_url" />
-              {{ option.full_name }}
-            </div>
-          </template>
-        </LabeledSelect>
-      </div>
-      <div class="col span-4">
-        <LabeledSelect
-          :disabled="!selectedRepo || loadingBranches"
-          :placeholder="branchPlaceholder"
-          :options="branches"
-          label="name"
-          :value="selectedBranch"
-          :clearable="false"
-          @input="selectBranch"
-        >
-        </LabeledSelect>
-      </div>
-      <div class="col span-4">
-        <LabeledSelect
-          :disabled="!selectedBranch"
-          :placeholder="filePlaceholder"
-          :options="files"
-          label="path"
-          :value="selectedFile"
-          :clearable="false"
-          @input="selectFile"
-        >
-        </LabeledSelect>
+      <div class="spacer archive">
+        <template>
+          <div class="spacer archive">
+            <LabeledInput
+              v-model="selectedUsername"
+              data-testid="epinio_app-source_git-username"
+              :tooltip="t('epinio.applications.steps.source.github.username.tooltip')"
+              :label="t('epinio.applications.steps.source.github.username.inputLabel')"
```

Comment:
as this is a generic component the l10n should live in the root dashboard en-us file